Mailbox.Tests/DataLoaderTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Mailbox.Tests
+{
+    [TestClass]
+    public class DataLoaderTests
+    {
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void DataLoader_Load_StreamIsNull_ThrowNullExeption()
+        {
+            DataLoader dL = new DataLoader(null);
+        }
+
+        [TestMethod]
+        public void DataLoader_Load_NotValidJson_ReturnsNull()
+        {
+            string filePath = Path.GetRandomFileName();
+            Stream stream = GenerateTextFile(filePath);
+            DataLoader dL = new DataLoader(stream);
+            List<Mailbox> worked = dL.Load();
+
+            stream.Close();
+            File.Delete(filePath);
+            Assert.IsNull(worked);
+        }
+
+        [TestMethod]
+        public void DataLoader_Load()
```

Comment:
You might want the name of your test(s) to consist of three parts:
- The name of the method being tested.
- The scenario under which it's being tested.
- The expected behavior when the scenario is invoked.
```suggestion
        public void Load_GeneratedFile_ReturnsCorrectStringRepresentation()
```